tensorflow/core/kernels/mkl_qmatmul_op_test.cc

```diff
@@ -471,6 +471,92 @@ TEST_F(QuantizedMatMulTest, Small_withBiasAndReluAndReq) {
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Attr("Toutput", DataTypeToEnum<qint32>::v())
+          .Attr("T", DataTypeToEnum<qint32>::v())
+          .Attr("_kernel", "QuantizedMklOp")
+          .Finalize(node_def()));
+  TF_ASSERT_OK(InitOp());
+  // The tensor shape of (1,3) is selected to allow the mkldnn expected
+  // weight format to be made as OI rather than IO for BS > 1
+  // A matrix is:
+  // |  1 |  2 |  3 |
+  AddInputFromArray<quint8>(TensorShape({1, 3}), {1, 2, 3});
+  // B matrix is:
+  // |  7 |  8 |  9 | 10 |
+  // | 11 | 12 | 13 | 14 |
+  // | 15 | 16 | 17 | 18 |
+  AddInputFromArray<qint8>(TensorShape({3, 4}),
+                           {7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18});
+  AddInputFromArray<qint32>(TensorShape({4}), {1, 2, 3, 4});
+  AddInputFromArray<float>(TensorShape({1}), {0});
+  AddInputFromArray<float>(TensorShape({1}), {255.0f});
+  AddInputFromArray<float>(TensorShape({1}), {-127.0f});
+  AddInputFromArray<float>(TensorShape({1}), {127.0f});
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+
+  TF_ASSERT_OK(RunOpKernel());
+  // Here are the results we expect, from hand calculations:
+  // (1 * 7) + (2 * 11) + (3 * 15) = 74
+  // (1 * 8) + (2 * 12) + (3 * 16) = 80
+  // (1 * 9) + (2 * 13) + (3 * 17) = 86
+  // (1 * 10) + (2 * 14) + (3 * 18) = 92
+  // Final result after Bias addition:
+  // 74  + 1 = 75 , 80  + 2 = 82 , 86  + 3 = 89 , 92  + 4 = 96,
+  Tensor expected(allocator(), DT_QINT32, TensorShape({1, 4}));
+  test::FillValues<qint32>(&expected, {75, 82, 89, 96});
+
+  const Tensor& output = *GetOutput(0);
+  const Tensor& mkl_shape_tensor = *GetOutput(3);
+  ConvMklToTF conv_comp;
+  Tensor output_quantized;
+  conv_comp.ConvertMKL2TF<qint32>(DT_QINT32, output, mkl_shape_tensor,
+                                  output_quantized);
+
+  test::ExpectTensorEqual<qint32>(expected, output_quantized);
+
+  // Test for the second time to use the cached weight
+  // Though we rely on performance as well as mkldnn verbose to
+  // determine if the weights buffer is cached and reused.
+  TF_ASSERT_OK(RunOpKernel());
```

Comment:
Can we time both calls and check (here) that the second call is faster?

Reply:
Added code as commented.

Reply:
Thank you very much!